controller/lib/core/comms.cpp

```diff
@@ -1,198 +1,31 @@
 #include "comms.h"
+
+#include "alg.h"
 #include "hal.h"
 #include "network_protocol.pb.h"
-#include "serdes.h"
+#include <pb_common.h>
+#include <pb_decode.h>
+#include <pb_encode.h>
 
-#define PACKET_LEN_MAX (32)
-static uint8_t tx_buffer[PACKET_LEN_MAX];
-static uint16_t tx_data_length;
-static bool output_buffer_ready = false;
+// TODO: Change last_{tx,rx}_ms to uint64 once Hal.millis() is uint64_t.
+//
+static uint8_t tx_buffer[ControllerStatus_size];
```

Comment:
Could you add a bit of comments as to what happens to the rx and tx buffers? Are they ring buffers, for example? Are they filled in wholly or incrementally?

Reply:
Done.